libs/design-system/src/color-scheme/useColorScheme.ts

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useLocalThemePreference } from '@novu/shared-web';
+import { useColorScheme as useMantineColorScheme } from '@mantine/hooks';
+import { ColorScheme } from './ColorScheme';
+import { mapThemeStatusToColorScheme } from './mapThemeStatusToColorScheme';
+
+/**
+ * Handle behavior for changing ColorSchemes or ThemeStatuses
+ */
+export const useColorScheme = () => {
```

Comment:
@BiswaViraj I ended up needing to modify the behavior of setting the color scheme, so I extracted it to this hook. It should still work with your "track" changes, but I will integrate it with your hook when we start bring our branches together